### ISSUE TITLE
[question] How to use`--require-override`?

### ISSUE THREAD

**Issue by madebr** [conan-io/conan#10010]
Conan 1.39.0 added the `--require-override` argument.
How can I use it?

Let's create the following `conanfile.txt`:
```
[requires]
cubicinterpolation/0.1.4
```
Current cubicinterpolation has a dependency on `boost/1.75.0`.
Executing the following command does not bump the version of boost.
It remains at 1.75.0.

```
conan install .. --require-override=boost/1.77.0
```

It's unclear to me whether this is a bug or whether I use this command wrongly?

I used the same syntax as tested by this commit: https://github.com/conan-io/conan/commit/ae7732bea1ff21ee4f53e8c636558d6425a6b999
Also, the test only tests `conanfile.py` recipes, no `conanfile.txt` recipes.

**UPDATE**:
It works when I convert `conanfile.txt` to `conanfile.py`.
So imho this is a bug: conanfile.txt has a different behavior then conanfile.py.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

Tested with conan 1.40.4

**Comment by franramirez688** [comment]
Hi @madebr 

Yes, it looks like a bug. It's behaving differently between each mode.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/10013 fixed this, will be in 1.43
